include/circt/Dialect/MSFT/MSFTPDOps.td

```diff
@@ -52,10 +58,34 @@ def PDRegPhysLocationOp : MSFTOp<"pd.reg_location",
   let assemblyFormat = [{
     (`ref` $ref^)? custom<ListOptionalRegLocList>($locs) attr-dict
   }];
+
+  let extraClassDeclaration = [{
+    Operation* getTopModule(circt::hw::HWSymbolCache &symCache) {
+      return getHierPathTopModule(getOperation()->getLoc(), symCache, getPathSym());
+    }
+  }];
+}
+
+def PDMulticycleOp : MSFTOp<"pd.multicycle",
```

Comment:
Nit: dunno if I'd consider a multi-cycle path a PD optimization. If it just generates tcl (as this appears to do), it's needed for correctness (I think), in which case it's definitely not a PD optimization. Just a necessary tcl command which is being abstracted here.

I would suggest (longer term) creating a MCP op in the pipeline dialect which both results in tcl and omits the registers.